homeassistant/components/supla/__init__.py

```diff
@@ -107,25 +141,38 @@ def discover_devices(hass, hass_config):
                 continue
 
             channel["server_name"] = server_name
-            component_configs.setdefault(component_name, []).append(channel)
+            component_configs.setdefault(component_name, []).append(
+                {
+                    "channel_id": channel_id,
+                    "server_name": server_name,
+                    "function_name": channel["function"]["name"],
+                }
+            )
 
     # Load discovered devices
-    for component_name, channel in component_configs.items():
-        load_platform(hass, component_name, "supla", channel, hass_config)
+    for component_name, config in component_configs.items():
+        await async_load_platform(hass, component_name, DOMAIN, config, hass_config)
```

Comment:
This is risking a deadlock between the platform setup and the integration setup. We should create tasks to set up the platforms. Please use `hass.async_create_task`.